Fix nightly test for EXO

ganglyu opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/sonic-net/SONiC/blob/gh-pages/CONTRIBUTING.md

Please provide following information to help code review process a bit easier:
-->
### Description of PR
<!--
- Please include a summary of the change and which issue is fixed.
- Please also include relevant motivation and context. Where should reviewer start? background context?
- List any dependencies that are required for this change.
-->

Summary:
Fix traffic issue on fanout switch.
Fixes # (issue)

### Type of change

<!--
- Fill x for your type of change.
- e.g.
- [x] Bug fix
-->

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)


### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [ ] 202305

### Approach
#### What is the motivation for this PR?
Nightly test for EXO is broken.
Microsoft ADO: 25579175

#### How did you do it?
1. Do not use duthost directly
2. Fix mac conflict on fanout switch

#### How did you verify/test it?
Run end to end test

#### Any platform specific information?

#### Supported testbed topology if it's a new test case?

### Documentation
<!--
(If it's a new feature, new test case)
Did you update documentation/Wiki relevant to your implementation?
Link to the wiki page?
-->

bingwang-ms commented:
Please add `PR summary` and `Type of change`.
Why do we need to enable this test on dualtor?

ganglyu commented:
> Please add `PR summary` and `Type of change`. Why do we need to enable this test on dualtor?

I have updated. I have fixed a dualtor related bug, but we don't need to enable this test on dualtor.

bingwang-ms commented:
> > Please add `PR summary` and `Type of change`. Why do we need to enable this test on dualtor?
> 
> I have updated. I have fixed a dualtor related bug, but we don't need to enable this test on dualtor.

If we don't need to enable this test on dualtor, why do we need this change?
```
Support dualtor
```

ganglyu commented:
> > > Please add `PR summary` and `Type of change`. Why do we need to enable this test on dualtor?
> > 
> > 
> > I have updated. I have fixed a dualtor related bug, but we don't need to enable this test on dualtor.
> 
> If we don't need to enable this test on dualtor, why do we need this change?
> 
> ```
> Support dualtor
> ```

It's a bug fix, we should not use duthost with rand_selected_dut

bingwang-ms commented on tests/acl/test_acl_outer_vlan.py:
```diff
@@ -557,7 +529,11 @@ def _do_verification(self, ptfadapter, duthost, tbinfo, vlan_setup_info, ip_vers
-        dst_mac = test_setup_config.get('dst_mac', ptfadapter.dataplane.get_mac(0, dst_port[0]))
+        if stage == INGRESS:
+            # Use broadcast for ingress test
+            dst_mac = "ff:ff:ff:ff:ff:ff"
```
Can you please clarify why broadcast MAC is used for ingress test?

bingwang-ms commented:
> > > > Please add `PR summary` and `Type of change`. Why do we need to enable this test on dualtor?
> > > 
> > > 
> > > I have updated. I have fixed a dualtor related bug, but we don't need to enable this test on dualtor.
> > 
> > 
> > If we don't need to enable this test on dualtor, why do we need this change?
> > ```
> > Support dualtor
> > ```
> 
> It's a bug fix, we should not use duthost with rand_selected_dut, and I have updated the description.

Discussed offline, let's keep using rand_selected_dut. It's the same with duthost on single tor testbed.

mssonicbld commented:
Cherry-pick PR to 202305: https://github.com/sonic-net/sonic-mgmt/pull/10568
